### ISSUE TITLE
ST11 (structure.unused_join) false positives for `LEFT SEMI` and `LEFT ANTI` joins

### ISSUE THREAD

**Issue by benberryallwood** [sqlfluff/sqlfluff#7886]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`sqlfluff lint` on a SQL file containing a `LEFT SEMI` join or `LEFT ANTI` join produces false positives for the [ST11 rule](https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-ST11).

Note that the inclusion of `LEFT` is optional and it's that which triggers the rule.

See [Spark SQL join types](https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-join.html#join-types).

### Expected Behaviour

No ST11 failure.

The ST11 docs say:

>Because some types of joins are often used as filters or to otherwise control granularity without being referenced (e.g. INNER and CROSS), this rule only applies to explicit OUTER joins (i.e. LEFT, RIGHT, and FULL joins).

SEMI and ANTI joins are _always_ "used as filters" so should also be excluded.

### Observed Behaviour

```
ST11 | Joined table 'tab2 AS r' not referenced elsewhere in query [structure.unused_join]
```

### How to reproduce

```sql
-- test.sql
SELECT l.id
FROM tab1 AS l
LEFT SEMI JOIN tab2 AS r
    ON l.id = r.id;
```

```sh
sqlfluff lint --dialect sparksql test.sql
```

### Dialect

`sparksql` and `databricks`

### Version

```
sqlfluff, version 4.2.1
Python 3.13.9
```

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
